nbconvert/exporters/slides.py

```diff
@@ -1,28 +1,78 @@
 """HTML slide show Exporter class"""
 
-#-----------------------------------------------------------------------------
-# Copyright (c) 2013, the IPython Development Team.
-#
+# Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
-#
-# The full license is in the file COPYING.txt, distributed with this software.
-#-----------------------------------------------------------------------------
 
-#-----------------------------------------------------------------------------
-# Imports
-#-----------------------------------------------------------------------------
+from copy import deepcopy
 
-from nbconvert import preprocessors
 from traitlets.config import Config
+from traitlets import Unicode
 
 from .html import HTMLExporter
 
-#-----------------------------------------------------------------------------
-# Classes
-#-----------------------------------------------------------------------------
+def prepare(nb):
+    """Add some convenenience metadata on cells for the slide template.
+    """
+    nb = deepcopy(nb)
+
+    for cell in nb.cells:
+        # Make sure every cell has a slide_type
+        cell.metadata.slide_type = cell.metadata.get('slideshow', {}).get('slide_type', '-')
+
+    # Find the first visible cell
+    for index, cell in enumerate(nb.cells):
+        if cell.metadata.slide_type not in {'notes', 'skip'}:
+            cell.metadata.slide_type = 'slide'
```

Comment:
I think conversion is the wrong time to warn people - it would be better to give them information like that in the interface when they're using the slide toolbar. But that's obviously much trickier to design.

Additionally, it doesn't really matter whether they start the slideshow 'properly' with a slide cell - we don't lose anything by accepting any visible cell as the start of the first slide (and RISE [does the same](https://github.com/damianavila/RISE/blob/56eff79aac0a0c49ad212c6944ff2fd16c774b65/livereveal/main.js#L71)).